frontend/views/containers/payments/PaymentRowReceived.vue

```diff
@@ -111,6 +117,14 @@ export default ({
         console.error(e)
         alert(e.message)
       }
+    },
+    async updatePaymentRelativeTo (paymentDate) {
+      this.relativeTo = await this.historicalPeriodStampGivenDate(paymentDate)
+    }
+  },
+  watch: {
+    payment (to, from) {
+      this.updatePaymentRelativeTo(to.date)
```

Comment:
Do we really need a watcher for this? I don't think it's necessary because AFAIK a `PaymentRowReceived` never has its `:payment` binding changed after being created.

For such instances it's simpler to initialize the value in the `mounted` method (which gets automatically called). That will also allow us to remove `updatePaymentRelativeTo` and directly calling `historicalPeriodStampGivenDate` in `mounted`.